File: front-end/src/types/domain/member-dashboard.ts
import type { MemberSessionsDto } from "./session";

export type MonthlyActivity = {
  month: string;
  numberOfAttendedSessions: number;
};

export type MemberDashboard = {
  joinedBefore: number;
  activeSubscriptions: number;
  monthAttendance: number;
  totalSpend: number;
  recentSessions: MemberSessionsDto[];
  upcomingSessions: MemberSessionsDto[];
  monthlyActivity: MonthlyActivity[];
};

File: IronCore/DTOs/MemberDashboard.cs
namespace IronCore.DTOs
{
    public class MemberDashboard
    {
        public required int JoinedBefore {  get; set; }
        public required int ActiveSubscriptions { get; set; }
        public required int MonthAttendance { get; set; }
        public required decimal TotalSpend { get; set; }
        public List<MonthlyActivity> MonthlyActivity { get; set; }
         = new List<MonthlyActivity>();
        public List<MemberSessionsDto> UpcomingSessions { get; set; }
         = new List<MemberSessionsDto>();
        public List<MemberSessionsDto> RecentSessions { get; set; }
         = new List<MemberSessionsDto>();
    }
}
